config/sources/common.conf

```diff
@@ -6,19 +6,180 @@
+		fi
+	fi
+
+	# KERNEL_MAJOR_MINOR is the key to all this stuff. It must be set.
+	if [[ -z ${KERNEL_MAJOR_MINOR} ]]; then
+		display_alert "KERNEL_MAJOR_MINOR is unset" "common_defaults_for_mainline" "warn"
+		((problems++))
+	fi
+
+	# exit with error if some breaking problem detected above.
+	if ((problems > 0)); then
+		exit_with_error "Please fix the above kernel configuration problems (check warnings above)."
+	fi
+
+	# Small debuggings to make sure we're not doing something stupid.
+	if [[ -z ${KERNELSOURCE} ]]; then
+		display_alert "KERNELSOURCE is unset" "common_defaults_for_mainline" "info"
+	else
+		display_alert "KERNELSOURCE is set by board/family/hooks to '${KERNELSOURCE}'" "common_defaults_for_mainline" "ext"
+	fi
+
+	if [[ -z ${KERNELBRANCH} ]]; then
+		display_alert "KERNELBRANCH is unset" "common_defaults_for_mainline" "info"
+	else
+		display_alert "KERNELBRANCH is set by board/family/hooks to '${KERNELBRANCH}'" "common_defaults_for_mainline" "ext"
+	fi
+
+	if [[ -z ${LINUXFAMILY} ]]; then
+		display_alert "LINUXFAMILY is unset" "common_defaults_for_mainline" "info"
+	else
+		display_alert "LINUXFAMILY is set by board/family/hooks to '${LINUXFAMILY}'" "common_defaults_for_mainline" "ext"
+	fi
+
+	if [[ -z ${KERNELPATCHDIR} ]]; then
+		display_alert "KERNELPATCHDIR is unset" "common_defaults_for_mainline" "info"
+	else
+		display_alert "KERNELPATCHDIR is set by board/family/hooks to '${KERNELPATCHDIR}'" "common_defaults_for_mainline" "ext"
+	fi
+
+	if [[ -z ${KERNEL_PATCH_ARCHIVE_BASE} ]]; then
+		display_alert "KERNEL_PATCH_ARCHIVE_BASE is unset" "common_defaults_for_mainline" "info"
+	else
+		display_alert "KERNEL_PATCH_ARCHIVE_BASE is set by board/family/hooks to '${KERNELPATCHDIR}'" "common_defaults_for_mainline" "ext"
+	fi
+
+	if [[ -z ${LINUXCONFIG} ]]; then
+		display_alert "LINUXCONFIG is unset" "common_defaults_for_mainline" "info"
+	else
+		display_alert "LINUXCONFIG is set by board/family/hooks to '${LINUXCONFIG}'" "common_defaults_for_mainline" "ext"
+	fi
+
+	# Generic defaultings, used for all kernels (including vendor/legacy/3rd-party and mainline).
+	# If KERNEL_PATCH_ARCHIVE_BASE is unset, set it to LINUXFAMILY.
+	if [[ -z ${KERNEL_PATCH_ARCHIVE_BASE} ]]; then
+		display_alert "KERNEL_PATCH_ARCHIVE_BASE is unset; using LINUXFAMILY: '${LINUXFAMILY}'" "common_defaults_for_mainline" "info"
+		KERNEL_PATCH_ARCHIVE_BASE="$LINUXFAMILY"
+	fi
+
+	# If KERNELPATCHDIR is unset, set it to "archive/${KERNELPATCHDIR}-${KERNEL_MAJOR_MINOR}"
+	if [[ -z ${KERNELPATCHDIR} ]]; then
+		display_alert "KERNELPATCHDIR is unset; using 'archive/${KERNEL_PATCH_ARCHIVE_BASE}-${KERNEL_MAJOR_MINOR}'" "common_defaults_for_mainline" "info"
+		KERNELPATCHDIR="archive/${KERNEL_PATCH_ARCHIVE_BASE}-${KERNEL_MAJOR_MINOR}" # previously was KERNELPATCHDIR="$LINUXFAMILY-$BRANCH"
+	fi
+
+	# If LINUFAMILY is unset... bomb
+	if [[ -z ${LINUXFAMILY} ]]; then
+		exit_with_error "LINUXFAMILY is unset after family config; KERNELSOURCE=${KERNELSOURCE}"
```

Comment:
You are using LINUXFAMILY to set KERNEL_PATCH_ARCHIVE_BASE on line 107. Hence this check needs to be before that check.

Reply:
Good catch. Will fix. Thanks

Reply:
Fixed in latest push, thanks